libchisel/src/dropsection.rs

```diff
@@ -44,6 +44,7 @@ fn names_section_index_for(module: &Module) -> Option<usize> {
 fn custom_section_index_for(module: &Module, name: &str) -> Option<usize> {
     module.sections().iter().position(|e| match e {
         Section::Custom(_section) => _section.name() == name,
+        Section::Name(_) => name == "name", // A name section is a custom section, but parity-wasm makes a distinction.
```

Comment:
The reasons are explained here: https://github.com/wasmx/wasm-chisel/blob/master/libchisel/src/utils.rs#L21